addons/account/wizard/account_move_reversal.py

```diff
@@ -26,8 +26,42 @@ class AccountMoveReversal(models.TransientModel):
             ('modify', 'Full refund and new draft invoice')
         ], string='Credit Method', required=True,
         help='Choose how you want to credit this invoice. You cannot "modify" nor "cancel" if the invoice is already reconciled.')
-    journal_id = fields.Many2one('account.journal', string='Use Specific Journal', help='If empty, uses the journal of the journal entry to be reversed.', check_company=True)
+    journal_id = fields.Many2one(
+        'account.journal',
+        check_company=True,
+        default=lambda self: self._get_default_journal_id(),
+        help='If empty, uses the journal of the journal entry to be reversed.',
+        required=True,
+        string='Use Specific Journal',
+    )
     company_id = fields.Many2one('res.company', required=True, readonly=True)
+    available_journal_ids = fields.Many2many('account.journal', compute='_compute_available_journal_ids')
+
+    @api.depends('move_ids')
+    def _compute_available_journal_ids(self):
+        for record in self:
+            if self.move_ids:
+                record.available_journal_ids = self.env['account.journal'].search([
+                    ('company_id', '=', self.company_id.id),
+                    ('type', 'in', self.move_ids.journal_id.mapped('type')),
+                ])
+            else:
+                record.available_journal_ids = self.env['account.journal'].search([('company_id', '=', self.company_id.id)])
+
+    def _get_default_journal_id(self):
+        journal = None
+        if self.env.context.get('active_model') == 'account.move':
+            ids = self.env.context.get('active_ids') or self.env.context.get('active_id')
+            moves = self.env['account.move'].browse(ids)
+            if len(moves.journal_id) == 1:
+                journal = moves.journal_id
+        return journal if journal and journal.active else None
+
+    @api.constrains('journal_id', 'move_ids')
+    def _check_journal_type(self):
+        for record in self:
+            if record.journal_id and record.move_ids and record.journal_id.type not in record.move_ids.journal_id.mapped('type'):
```

Comment:
`if record.journal_id.type not in record.move_ids.journal_id.mapped('type'):` is enough